pylint/checkers/deprecated.py

```diff
@@ -73,6 +78,27 @@ def visit_import(self, node):
                 mod_name, class_name = name.split(".", 1)
                 self.check_deprecated_class(node, mod_name, (class_name,))
 
+    def deprecated_decorators(self) -> Container[str]:
+        """Callback returning the deprecated decorators.
+
+        Returns:
+            collections.abc.Container of deprecated decorator names.
+        """
+        # pylint: disable=no-self-use
+        return ()
+
+    @utils.check_messages(
+        "deprecated-decorator",
+    )
+    def visit_decorators(self, node):
+        """Triggered when a decorator statement is seen"""
+        if isinstance(next(node.get_children()), astroid.Call):
+            qname = next(node.get_children()).func.inferred()[0].qname()
+        else:
+            qname = next(next(node.get_children()).infer()).qname()
```

Comment:
I'm not sure you can be certain that the result will always have a `qname()` method here.

Reply:
Instead of `.infer()` try `safe_infer` from the `pylint/checkers/utils.py` module.